src/System.Net.Primitives/src/System/Net/IPv6AddressHelper.cs

```diff
@@ -0,0 +1,472 @@
+                                // the array if there's a prefix
+                                //
+
+                                number = 0;
+                                numberIsValid = false;
+                                break;
+                            }
+                        }
+                        break;
+
+                    case '/':
+                        if (numberIsValid)
+                        {
+                            numbers[index++] = (ushort)number;
+                            numberIsValid = false;
+                        }
+
+                        //
+                        // since we have a valid IPv6 address string, the prefix
+                        // length is the last token in the string
+                        //
+
+                        for (++i; address[i] != ']'; ++i)
+                        {
+                            PrefixLength = PrefixLength * 10 + (address[i] - '0');
+                        }
+                        break;
+
+                    default:
+                        number = number * 16 + Uri.FromHex(address[i++]);
+                        break;
+                }
+            }
+
+            //
+            // add number to the array if its not the prefix length or part of
+            // an IPv4 address that's already been handled
+            //
+
+            if (numberIsValid)
+            {
+                numbers[index++] = (ushort)number;
+            }
+
+            //
+            // if we had a compressor sequence ("::") then we need to expand the
+            // numbers array
+            //
+
+            if (compressorIndex > 0)
+            {
+
+                int toIndex = NumberOfLabels - 1;
+                int fromIndex = index - 1;
+
+                for (int i = index - compressorIndex; i > 0; --i)
+                {
+                    numbers[toIndex--] = numbers[fromIndex];
+                    numbers[fromIndex--] = 0;
+                }
+            }
+
+            //
+            // is the address loopback? Loopback is defined as one of:
+            //
+            //  0:0:0:0:0:0:0:1
+            //  0:0:0:0:0:0:127.0.0.1       == 0:0:0:0:0:0:7F00:0001
+            //  0:0:0:0:0:FFFF:127.0.0.1    == 0:0:0:0:0:FFFF:7F00:0001
+            //
+
+            return ((numbers[0] == 0)
+                            && (numbers[1] == 0)
+                            && (numbers[2] == 0)
+                            && (numbers[3] == 0)
+                            && (numbers[4] == 0))
+                           && (((numbers[5] == 0)
+                                && (numbers[6] == 0)
+                                && (numbers[7] == 1))
+                               || (((numbers[6] == 0x7F00)
+                                    && (numbers[7] == 0x0001))
+                                   && ((numbers[5] == 0)
+                                       || (numbers[5] == 0xFFFF))));
+
```

Comment:
It seems like we don't actually use the return value here.  Did I miss some call site that uses this?